src/316C00.c

```diff
@@ -1,3 +1,62 @@
 #include "common.h"
+#include "nu/nusys.h"
+#include "ld_addrs.h"
 
-INCLUDE_ASM(s32, "316C00", func_802AE000_316C00);
+#define JAL   0x03
+#define ADDIU 0x09
+#define LUI   0x0F
+#define LW    0x23
+#define LOWER(x) ((x) & 0xFFFF)
+#define UPPER(x) ((x) >> 16)
+#define OPCODE(x) ((x) >> 26)
+#define GET_RS(x) ((x >> 0x15) & 0x1F)
+#define GET_RT(x) (UPPER(x) & 0x1F)
+
+extern s32 D_7012BC11[];
+
+void func_802AE000_316C00(void) {
+    s32(*readFunc)(OSPiHandle*, u32, u32*);
+    UNK_FUN_PTR(generalHeapCreate);
+    u32 thisInsn;
+    u32* it;
+    u32 prevInsn;
+    s32 seed;
+    u32 blah;
+    u32 hash;
+
+    readFunc = osEPiReadIo;
+    seed = 0x4BF15BF6; // 0x3C016C07 + 0xFEFEFEF;
```

Comment:
Does it not match to just evaluate the expression at compile-time, like
```suggestion
    seed = 0x3C016C07 + 0xFEFEFEF;
```

Reply:
It does, but I thought it's useful to show both the actual value and how it's composed in the comment